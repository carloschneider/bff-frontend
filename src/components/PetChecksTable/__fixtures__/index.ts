import { CheckType } from '../graphql'

export const getAllChecksByPetId: CheckType[] = [
  {
    id: 'a6a3300f-fcf8-40cc-89f2-1fa85d4c4ba2',
    arrive: '2023-01-25T18:26:43.000Z',
    leave: '2023-01-25T18:27:36.000Z',
    count: 6,
    responsibles: [
      {
        id: '9a6f4710-fd3e-49ac-a0f8-d38498e7b45b',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '62ad42eb-699c-4b8a-a0a3-c0f25189acaa',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: '4b63fd9c-d49b-4c5d-8c38-1fe72a6e2f88',
    arrive: '2023-01-07T17:31:36.000Z',
    leave: '2023-01-07T17:31:38.000Z',
    count: null,
    responsibles: [
      {
        id: 'cdee01c6-044a-47a3-a0f5-8b94fa59ee3d',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: 'b9d79e38-bc04-4674-b098-ee00f858e57d',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: 'ade1d25c-0802-4aca-8524-0f493a609dc2',
    arrive: '2023-01-07T17:29:39.000Z',
    leave: '2023-01-07T17:29:41.000Z',
    count: null,
    responsibles: [
      {
        id: 'b82f9601-1660-4f0a-a19c-f10237868916',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '6eea4caa-445d-42b8-b196-dc9fdf25f1f8',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: '8846501b-c64d-4708-85e1-ba028e02430c',
    arrive: '2023-01-07T03:35:02.000Z',
    leave: '2023-01-07T03:35:04.000Z',
    count: null,
    responsibles: [
      {
        id: 'be15638d-f468-4f25-837d-a384c37c2464',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '18be9589-8c4e-4dd6-9fa2-ef5a0a4f2876',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  }
]

export const getAllChecksByPetIdPageTwo: CheckType[] = [
  {
    id: 'ade1d25c-0802-4aca-8524-0f493a609dc2',
    arrive: '2023-01-07T17:29:39.000Z',
    leave: '2023-01-07T17:29:41.000Z',
    count: 8,
    responsibles: [
      {
        id: 'b82f9601-1660-4f0a-a19c-f10237868916',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '6eea4caa-445d-42b8-b196-dc9fdf25f1f8',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: '8846501b-c64d-4708-85e1-ba028e02430c',
    arrive: '2023-01-07T03:35:02.000Z',
    leave: '2023-01-07T03:35:04.000Z',
    count: 7,
    responsibles: [
      {
        id: 'be15638d-f468-4f25-837d-a384c37c2464',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: '18be9589-8c4e-4dd6-9fa2-ef5a0a4f2876',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: 'e0d3a00b-56d9-4667-abd6-edb62ed1a746',
    arrive: '2023-01-03T21:47:23.000Z',
    leave: '2023-01-03T21:47:32.000Z',
    count: null,
    responsibles: [
      {
        id: '0eaaf655-86c5-4ff6-946b-5fa94f1747ae',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: 'be9d898c-e7d9-45ca-ac19-b2824248cb98',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  },
  {
    id: 'd34065a1-53b7-40b4-a949-8c21268461bb',
    arrive: '2023-01-03T20:32:56.000Z',
    leave: '2023-01-03T20:32:58.000Z',
    count: null,
    responsibles: [
      {
        id: 'bfae6391-9bcd-403a-a54f-ea22066b8dfe',
        type: 'arrive',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      },
      {
        id: 'c2ad7b13-a5a0-4df6-9813-14300f83a880',
        type: 'leave',
        staff: {
          id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
          firstName: 'Silvio',
          lastName: 'Santos'
        },
        user: null
      }
    ]
  }
]
