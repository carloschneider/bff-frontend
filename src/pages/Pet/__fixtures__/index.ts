import { PetType } from '../graphql'

export const getPetByIdFixture: PetType = {
  id: 'fake-id',
  name: 'Fake Pet Name',
  birthDate: new Date('2018-11-11T01:00:00.000Z'),
  breed: {
    name: 'Fake Pet Breed'
  },
  tutors: [
    {
      id: 'fake-id',
      firstName: 'Fake',
      lastName: 'Tutor',
      phoneNumber: '(99) 99999-9999'
    }
  ]
}
