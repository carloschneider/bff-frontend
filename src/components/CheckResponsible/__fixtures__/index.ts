import { CheckType } from 'components/PetChecksTable/graphql'

export const checkResponsibleArrive: CheckType = {
  id: 'a6a3300f-fcf8-40cc-89f2-1fa85d4c4ba2',
  arrive: '2023-01-25T18:26:43.000Z',
  leave: null,
  count: 6,
  responsibles: [
    {
      id: '9a6f4710-fd3e-49ac-a0f8-d38498e7b45b',
      type: 'arrive',
      staff: {
        id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
        firstName: 'Silvio',
        lastName: 'Santos'
      },
      user: null
    }
  ]
}

export const checkResponsibleLeave: CheckType = {
  id: '4b63fd9c-d49b-4c5d-8c38-1fe72a6e2f88',
  arrive: '2023-01-07T17:31:36.000Z',
  leave: '2023-01-07T17:31:38.000Z',
  count: 5,
  responsibles: [
    {
      id: 'b9d79e38-bc04-4674-b098-ee00f858e57d',
      type: 'arrive',
      staff: null,
      user: {
        id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
        companyName: 'Walky Reabilitação'
      }
    },
    {
      id: 'cdee01c6-044a-47a3-a0f5-8b94fa59ee3d',
      type: 'leave',
      staff: {
        id: 'db8653e6-7ffe-4edf-aa36-8f26c11e8576',
        firstName: 'Fausto',
        lastName: 'Silva'
      },
      user: null
    }
  ]
}
