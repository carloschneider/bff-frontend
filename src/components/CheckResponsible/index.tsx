import { CheckType } from 'components/PetChecksTable/graphql'

type CheckRenderProps = {
  date: Date | null
  record: CheckType
  type: 'arrive' | 'leave'
}

const dateOptions: Intl.DateTimeFormatOptions = {
  hourCycle: 'h23'
}

const CheckResponsible = ({ date, record, type }: CheckRenderProps) => {
  const getResponsibleName = (): string => {
    return record.responsibles
      .filter((responsible) => responsible.type === type)
      .reduce((acc, responsible) => {
        if (responsible.user === null) {
          acc = `${responsible.staff?.firstName} ${responsible.staff?.lastName}`

          return acc
        }

        acc = responsible.user.companyName

        return acc
      }, '')
  }

  return (
    <>
      {date ? new Date(date).toLocaleString('pt-BR', dateOptions) : '-'}{' '}
      {date && `(${getResponsibleName()})`}
    </>
  )
}

export default CheckResponsible
