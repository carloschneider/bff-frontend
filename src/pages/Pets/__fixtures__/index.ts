import { PetType } from 'pages/Pet/graphql'

export const getAllPetsFixture: Omit<PetType, 'tutors'>[] = [
  {
    id: 'c504316d-5d49-4a8e-8d3f-0e47e94a8694',
    name: 'Fake Pet Name 1',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:40.000Z'),
    breed: {
      name: 'Fake Breed 1'
    },
    count: 2
  },
  {
    id: '54737ecb-ed7e-4f5a-b645-25e7033f5fc2',
    name: 'Fake Pet Name 2',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:38.000Z'),
    breed: {
      name: 'Fake Breed 2'
    },
    count: null
  }
]

export const getAllPetsOrderFixture: Omit<PetType, 'tutors'>[] = [
  {
    id: '54737ecb-ed7e-4f5a-b645-25e7033f5fc2',
    name: 'Fake Pet Name 2',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:38.000Z'),
    breed: {
      name: 'Fake Breed 2'
    },
    count: 2
  },
  {
    id: 'c504316d-5d49-4a8e-8d3f-0e47e94a8694',
    name: 'Fake Pet Name 1',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:40.000Z'),
    breed: {
      name: 'Fake Breed 1'
    },
    count: null
  }
]

export const getAllPetsSearchResultFixture: Omit<PetType, 'tutors'>[] = [
  {
    id: 'c504316d-5d49-4a8e-8d3f-0e47e94a8694',
    name: 'search pet',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:40.000Z'),
    breed: {
      name: 'Fake Breed Search'
    },
    count: 1
  }
]

export const getAllPetsPaginationFixture: Omit<PetType, 'tutors'>[] = [
  {
    id: 'c504316d-5d49-4a8e-8d3f-0e47e94a8694',
    name: 'Fake Pet Name 1',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:40.000Z'),
    breed: {
      name: 'Fake Breed 1'
    },
    count: 3
  },
  {
    id: '54737ecb-ed7e-4f5a-b645-25e7033f5fc2',
    name: 'Fake Pet Name 2',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:38.000Z'),
    breed: {
      name: 'Fake Breed 2'
    },
    count: null
  }
]

export const getAllPetsPaginationPageTwoFixture: Omit<PetType, 'tutors'>[] = [
  {
    id: 'a6a3300f-fcf9-40cc-89f2-1fa85d4c4ba2',
    name: 'Fake Pet Name 3',
    birthDate: new Date('2018-11-11T01:00:00.000Z'),
    createdAt: new Date('2022-12-30T17:45:40.000Z'),
    breed: {
      name: 'Fake Breed 3'
    },
    count: 3
  }
]
