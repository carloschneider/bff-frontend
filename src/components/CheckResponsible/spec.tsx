import { render, screen } from '@testing-library/react'

import { checkResponsibleLeave } from './__fixtures__'

import CheckResponsible from '.'

describe('<CheckResponsible />', () => {
  it('should render component', () => {
    const { container } = render(
      <CheckResponsible
        date={new Date()}
        type="arrive"
        record={checkResponsibleLeave}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should ', () => {
    const date = new Date()

    render(
      <CheckResponsible
        date={date}
        type="leave"
        record={checkResponsibleLeave}
      />
    )

    const responsible = screen.getByText(
      `${date.toLocaleString('pt-BR', {
        hourCycle: 'h23'
      })} (Fausto Silva)`
    )

    expect(responsible).toBeInTheDocument()
  })
})
