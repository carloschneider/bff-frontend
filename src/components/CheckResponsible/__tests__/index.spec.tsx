import { render, screen } from '@testing-library/react'

import CheckResponsible from '../'
import { checkResponsibleLeave } from '../__fixtures__'

describe('<CheckResponsible />', () => {
  it('should render component', () => {
    const { container } = render(
      <CheckResponsible
        date={new Date('2020-01-01T20:00:00.573Z')}
        type="arrive"
        record={checkResponsibleLeave}
      />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render date and staff member name', () => {
    const date = new Date()

    render(
      <CheckResponsible
        date={date}
        type="leave"
        record={checkResponsibleLeave}
      />
    )

    const responsible = screen.getByText(
      `${date.toLocaleString('pt-BR', {
        hourCycle: 'h23'
      })} (Fausto Silva)`
    )

    expect(responsible).toBeInTheDocument()
  })

  it('should show hyphen when data is null', () => {
    render(
      <CheckResponsible
        date={null}
        type="leave"
        record={checkResponsibleLeave}
      />
    )

    const hyphen = screen.getByText('-')

    expect(hyphen).toBeInTheDocument()
  })
})
