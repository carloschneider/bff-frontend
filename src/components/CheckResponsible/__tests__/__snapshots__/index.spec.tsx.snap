// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckResponsible /> should render component 1`] = `01/01/2020 17:00:00`;
